chalk-solve/src/clauses.rs

```diff
@@ -214,7 +214,11 @@ fn program_clauses_that_could_match(
         DomainGoal::LocalImplAllowed(trait_ref) => db
             .trait_datum(trait_ref.trait_id)
             .to_program_clauses(db, clauses),
-        DomainGoal::Compatible(()) => (),
+        DomainGoal::Compatible(()) | DomainGoal::RevealMode(()) => (),
+        DomainGoal::Overrides(ov) => {
```

Comment:
So here we need to find all the "sources" of overrides clauses. I think these would derive from lowering the "assoc type values" found in impl of the given trait, so the simplest thing would be to call `push_program_clauses_for_associated_type_values_in_impls_of` like we do for `Normalize`.